tools/autograd/templates/Functions.cpp

```diff
@@ -642,9 +642,20 @@ Tensor _fused_dropout_backward(Tensor grad, Tensor mask, double p1m) {
   }
 }
 
-Tensor select_equals_backward(Tensor grad, const Tensor & input, const Tensor & value) {
+Tensor select_first_equal_backward(Tensor grad, const Tensor & input, const Tensor & value) {
   auto grad_input = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
Talked to @VitalyFedyunin, it's ok to change the format of gradient here to match the format of input (that is, remove LEGACY_CONTIGUOUS argument, zeros_like would do the right thing automatically). It would also make sense to add a test for channels_last/permuted input in this case.

Reply:
channels_last/permuted? Could you please elaborate, I am not sure I understand :)

Reply:
If you have a non contiguous input (e.g. created by `torch.randn(2,3).t()`), in most cases you'd like the gradient to have the same strides. Same for channels-last tensor (`torch.randn(2,3,4,5)).contiguous(memory_format=torch.channels_last)`. That could be achieved by creating grad_input with at::zeros_like (w/o an additional argument). You can inspect the gradient that you produce with gradient hooks in the test. Unfortunately just looking at input's .grad attribute won't help because it will be made contiguous (long story, we'll fix it). Also, feel free to ignore it, really it's outside of scope for this PR.